### ISSUE TITLE
[BUG] Artifacts after CSGIntersection  Booleans

### ISSUE THREAD

**Issue by mifth** [AcademySoftwareFoundation/openvdb#1072]
### Environment
**Operating System:** Windows 10
**Version / Commit SHA:** 8.0
**Other:** 

### Describe the bug
There are artifacts after **csgIntersection booleans** if I convert voxels to mesh. I assume this is a bug as Union and Difference work without issues. But Intersection has an issue.
![image](https://user-images.githubusercontent.com/2251298/119398051-ebc87580-bcce-11eb-873a-43abfa3070be.png)

Here is my blender file where i loaded my meshes so that you could check it.
[tmp1.zip](https://github.com/AcademySoftwareFoundation/openvdb/files/6534611/tmp1.zip)

And Here are my vdb volumes for testing:
[vdb_volumes.zip](https://github.com/AcademySoftwareFoundation/openvdb/files/6534614/vdb_volumes.zip)

My code looks like this:
```
// load 2 volumes

openvdb::tools::csgIntersection(*gridCylinder, *gridSphere);

openvdb::tools::VolumeToMesh mesher(0, 0.0, true);
mesher.operator()<openvdb::FloatGrid>( gridCylinder.operator*() );
// then I get polygons and points
```

### To Reproduce
1. Load my 2 vdb files
2. Boolean them with **openvdb::tools::csgIntersection**
3. Convert to mesh with **openvdb::tools::VolumeToMesh**

### How I fixed it
I had to copy sphereGrid and parse combineExtended function taken from the CoockBook. But I believe I should not do this as intersection booleans **must work correctly**.

```
// Define a local function that retrieves a and b values from a CombineArgs
// struct and then sets the result member to the maximum of a and b.
struct Local {
    static inline void max(openvdb::CombineArgs<float>& args) {
        if (args.b() > args.a()) {
            // Transfer the B value and its active state.
            args.setResult(args.b());
            args.setResultIsActive(args.bIsActive());
        } else {
            // Preserve the A value and its active state.
            args.setResult(args.a());
            args.setResultIsActive(args.aIsActive());
        }
    }
};

openvdb::FloatGrid::Ptr tempGrid  = gridCylinder->deepCopy();
openvdb::tools::csgIntersection(*gridCylinder, *gridSphere);
gridCylinder->tree().combineExtended(tempGrid->tree(), Local::max);
```

Then I get nice Intersection booleans:
![image](https://user-images.githubusercontent.com/2251298/119400374-005a3d00-bcd2-11eb-8949-21ad088344aa.png)

**Comment by Idclip** [comment]
@mifth I believe this bug was introduced in VDB `8.0.0` and was fixed in `8.0.1`. Can you try you example with `8.0.1`?

**Comment by mifth** [comment]
> @mifth I believe this bug was introduced in VDB `8.0.0` and was fixed in `8.0.1`. Can you try your example with `8.0.1`?

I confirm this issue is fixed in `8.0.1`. Thank you for fast reply.
